### PR TITLE
feat(component): compound & numeric reactive_show predicates (#176)

### LINKED ISSUE #176: Compound & numeric reactive_show predicates: all:/any: across fields and gte:/gt:/lte:/lt: thresholds
## Summary

`reactive_show` (#161) and `reactive_show_targets` (#164) cover the `x-show` / `data-show` case beautifully **as long as visibility is decidable from ONE field with ONE literal predicate**. Two very common form shapes fall just outside that envelope today, and each still forces a hand-written `change`-listener Stimulus controller back into an otherwise fully-declarative reactive form — the exact thing these bindings exist to remove:

1. **Compound conditions** — visibility that depends on **more than one field**, or on **more than one value of one field combined with AND/OR**. e.g. "show the address block while `type == individual` **AND** `country != domestic`", or "show the name fields while `role == director` **OR** `role == shareholder`".
2. **Numeric / ordered comparisons** — visibility gated on a **threshold**, not a literal match. e.g. "show the surcharge notice while `quantity >= 10`", "show the large-order warning while `amount > 5000`".

The predicate vocabulary is deliberately literal-only (`equals:` / `not:` / `in:`, all stringified) — which is right for the eval-free security posture — but it means neither of the above is expressible, and both are extremely common.

### Compound (AND across two fields)

```ruby
# I want: show the block while type == "individual" AND country != "domestic".
# Each binding toggles its OWN `hidden` independently, so I can't express an
# AND on one element. I *can* fake AND by NESTING two bindings:
div(**reactive_show(:type, equals: "individual")) do
  div(**reactive_show(:country, not: "domestic")) do
    # ...address fields...
  end
end
# ...but this only works for AND, forces an extra wrapper div per term, and
# couples the DOM structure to the boolean logic. There is no OR at all.
```

### Compound (OR across two fields)

```ruby
# I want: show the name fields while director OR shareholder is checked.
# There is no primitive for this — nesting only gives AND, and two sibling
# copies of the same fields would emit duplicate `name=` inputs. So this
# single case drags a whole Stimulus controller back in.
```

### Numeric threshold

```ruby
# I want: show the surcharge notice while quantity >= 10.
# equals:/in: are literal string matches — there is no >=, >, <=, < .
div(**reactive_show(:quantity, equals: "10"))   # only the exact string "10" — wrong
div(**reactive_show(:quantity, in: %w[10 11 12 …])) # unbounded, absurd — wrong
```

`reactive_compute` reads field values and can derive numbers, but its outputs are field values / text mirrors — not a visibility toggle. So compound-and-numeric visibility falls through the same crack #161 originally described, just one layer up.

## Proposal (client-only, zero round trip, same eval-free posture)

Two additions, both staying within the "declared literal predicate, no expression eval" contract. Either can ship independently.

### A. Compound predicate — `all:` / `any:` over per-field literal terms

A binding may declare a list of `{ field, <predicate> }` terms combined with a single top-level connective. Each term is the SAME literal vocabulary (`equals:`/`not:`/`in:`); only the *combination* is new, so there is still no expression surface — the connective is one of two fixed keywords.

```ruby
# AND: visible while type == "individual" AND country != "domestic"
div(**reactive_show(all: [
  { field: :type,    equals: "individual" },
  { field: :country, not:    "domestic" }
]))

# OR: visible while role == "director" OR role == "shareholder"
# (here the two terms are the SAME field — subsumes a value-set, but any:
#  also spans DIFFERENT fields, which in: cannot)
div(**reactive_show(any: [
  { field: :director,    equals: true },
  { field: :shareholder, equals: true }
]))
```

Semantics: the client reads each term's owned field value, evaluates its literal predicate (the existing `showBindingMatches` logic per term), and `all:` ANDs / `any:` ORs the booleans, then toggles `hidden`. A single-field `all:`/`any:` with one term degrades to today's behaviour. Malformed term → that term is warn-and-skipped and treated as… (open question: `false` for `all:`, skip for `any:`? — see below). The single-predicate form (`reactive_show(:field, equals:)`) stays exactly as-is; `all:`/`any:` are additive and mutually exclusive with it (enforced loudly at render, like the current one-predicate rule).

This also cleanly replaces the nesting workaround: one flat binding on one element instead of N wrapper divs, and the DOM structure no longer encodes the boolean logic.

### B. Numeric predicates — `gte:` / `gt:` / `lte:` / `lt:`

Four new predicate keys that coerce **both** sides to Number and compare. Still a declared literal (the RHS is a literal number in the binding, never an expression); the only new capability is ordered comparison instead of equality.

```ruby
div(**reactive_show(:quantity, gte: 10))   # visible while Number(value) >= 10
div(**reactive_show(:amount,   gt:  5000)) # visible while Number(value) >  5000
```

Semantics: `const n = Number(value); return Number.isNaN(n) ? false : n >= literal`. A non-numeric field value (empty, blank, garbage) is `NaN` → predicate is `false` (hidden), which is the safe default for a "reveal on threshold" notice. Composes with A: `{ field: :amount, gte: 5000 }` as a term inside `all:`/`any:`.

## Scope / security

Deliberately stays inside the existing light, presentational envelope:

- **Client-only, no round trip, no token** — same category as today's `reactive_show`.
- **Reads OWNED fields only** — every term goes through the same `#ownsField` / ownership filter; `all:`/`any:` can only reference fields this root owns (same as the single-field form).
- **No expression eval** — the connective is one of two fixed keywords (`all`/`any`), each term is the existing literal vocabulary, and the numeric keys compare against a literal number baked into the attribute. There is no place a caller-supplied string is evaluated as code. Same two-sided default-deny: the Ruby helper validates shape loudly at render; the client warn-and-skips a malformed term rather than throwing or blanking the page.
- **`hidden` toggle only** — no `innerHTML`, no class/attribute freedom, exactly as now.
- Cross-root: `reactive_show_targets` could accept the same compound/numeric predicate objects in its per-id map, since it already carries a parsed predicate object per target — `showPredicateMatches` would grow the same `all`/`any`/numeric arms.

## Wire shape (sketch)

The single-predicate form stays as flat `data-reactive-show-*` attrs. The compound form needs a structured payload, so it serializes like `reactive_show_targets` already does — one JSON attribute:

```ruby
# data-reactive-show='{"all":[{"field":"type","equals":"individual"},{"field":"country","not":"domestic"}]}'
```

so `showBindingMatches` grows an `all`/`any` arm that recurses over terms (reusing the per-field value memo `#syncShow` already builds), and the numeric arm is four more `getAttribute`/key checks alongside `equals`/`not`/`in`.

## Open questions

1. **Malformed-term policy under `all:` vs `any:`** — treat a warn-skipped term as `false` (fail-closed, so `all:` hides and `any:` ignores it) or skip it entirely from the fold? Fail-closed matches the rest of the default-deny posture.
2. **Numeric coercion locale** — `Number()` is `.`-decimal only. A locale that types `1,5` would `NaN`. Probably fine (fields feeding a numeric predicate are `<input type=number>`), but worth stating.
3. **Should `gte:` etc. also accept a string** ("10") for symmetry with the stringified literal predicates, or require an actual `Numeric` in Ruby (stricter, catches typos at render)? Leaning stricter.

## Why build it in

"Reveal a section when a control has a certain value" — #161's own motivation — very frequently means *two* controls, or a *threshold*, in real forms: conditional fieldsets keyed on a type + a sub-choice, address blocks gated on residency, surcharge/warning notices gated on a quantity or amount, multi-role pickers where several checkboxes reveal a shared block. Each of those is one hand-written `change`-listener controller today, sitting inside an otherwise fully-declarative reactive form. The ownership + predicate plumbing all exists; A is a fold over the existing per-term evaluator, and B is four sibling predicate keys. Happy to help with a PR if the shape is agreed.

### PR DESCRIPTION
**Pull request by mhenrixon**
Closes #176.

## Summary

Extends value-conditional visibility so a client-side show/hide can depend on **more than one field** (`all:`/`any:`) and on **numeric thresholds** (`gte:`/`gt:`/`lte:`/`lt:`), staying inside the eval-free "declared literal predicate" contract. These were the two common form shapes that fell just outside `reactive_show` (#161) / `reactive_show_targets` (#164) and forced a hand-written `change`-listener Stimulus controller back into an otherwise fully-declarative reactive form.

### A. Compound — `all:` / `any:` over per-field terms

One fixed connective folds a list of `{ field:, <predicate> }` terms (same `equals:`/`not:`/`in:`/`gte:`/… vocabulary). No expression surface — the connective is one of two fixed keywords. A compound has no single controlling field, so it serializes as one JSON attr (`data-reactive-show`), like `reactive_show_targets`.

```ruby
# visible while type == "individual" AND country != "domestic"
div(**reactive_show(all: [
  { field: :type,    equals: "individual" },
  { field: :country, not:    "domestic" }
]))

# visible while director OR shareholder is checked
div(**reactive_show(any: [
  { field: :director,    equals: true },
  { field: :shareholder, equals: true }
]))
```

Replaces the wrapper-div nesting workaround (the DOM structure no longer encodes the boolean logic) and is the only way to express OR. Additive and mutually exclusive with the single-field form and with each other — raised loudly at render.

### B. Numeric — `gte:` / `gt:` / `lte:` / `lt:`

Compare `Number(value)` against a literal number baked into the binding. Works standalone, as a compound term, and inside a `reactive_show_targets` map.

```ruby
div(**reactive_show(:quantity, gte: 10))
div(**reactive_show(all: [{ field: :type, equals: "order" }, { field: :amount, gte: 5000 }]))
```

## Test Coverage

| Layer | Adds |
|-------|------|
| Ruby unit | numeric + compound helper cases: wire shape, `Numeric`-RHS validation, `0`/negative/float thresholds, connective mutual-exclusion, per-term validation — for `reactive_show` and `reactive_show_targets` |
| JS unit | numeric operators, `all:`/`any:` folds, numeric term inside a compound, malformed-payload skip, compound-only gate, and the `Number("") === 0` regression |
| System (browser) | compound AND + numeric threshold exercised for real; still **zero fetches, no reload**; green under Puma **and** Falcon |
| Docs | example page + live-showcase component updated; CHANGELOG `Added` entry |

## Verification

- `bundle exec rubocop` — clean (lib + spec + docs)
- `bundle exec rspec spec/phlex spec/requests` — 967 passed
- `bun test spec/javascript` — 360 passed
- `spec/system/conditional_fieldset_spec.rb` — green under Puma AND Falcon
- Vendored client re-synced (byte-identical, `vendored_controller_sync_spec` green); `rake build:js_check` clean
- pgbus optionality untouched (client-only feature — no transport change)
- Backwards compatible: the single-field `reactive_show(:field, equals:)` form is unchanged; every prior spec still passes

## Deviations & judgment calls

Read this first — it is the audit trail for every decision the issue didn't settle.

**Open questions resolved (per the issue's own lean):**
- **Q1 malformed-term policy → fail-closed.** A warn-skipped/missing/malformed term folds to `false`: `all:` hides (a broken AND term can't pass), `any:` ignores it (a broken OR term can't reveal). Matches the default-deny posture.
- **Q3 numeric RHS type → stricter.** `gte:` etc. require an actual `Numeric` in Ruby — a non-numeric literal raises at render (catches typos). The client still coerces the *field* value via `Number()`.

**Judgment calls & discoveries:**
- **Wire shape:** the single-field numeric form is a flat `data-reactive-show-gte="10"` (attr convention, `Number()`-coerced on read); the compound/targets embedded form keeps the RHS a real JSON number (`{"gte":5000}`). One shared `numericPredicateMatches` handles both.
- **Compound attr name** `data-reactive-show` (bare) — a compound has no single field, so the gate and sync walk use the combined selector `[data-reactive-show-field], [data-reactive-show]`.
- **A missing owned field in a compound term folds `false`** (fail-closed), whereas a single-field binding leaves visibility alone — a fold can't cleanly skip one term.
- **The docs live-example component is a separate copy** from the dummy; both were updated so the published demo shows the new sections and the system spec exercises them.
- **BUG caught by an adversarial review pass and FIXED before this PR:** `Number("")` / `Number("  ")` return `0`, not `NaN`. A blank field with `lte: 5` / `lt: 5` / `gte: 0` would have wrongly *revealed* (0 satisfies the comparison), violating the documented "blank → hidden" contract — my first guard only fail-closed for operators where `0` happened to fail (e.g. `gte: 10`). `numericPredicateMatches` now forces an empty/whitespace/null value to `NaN` before comparing, so **every** operator fails closed on a blank field. A dedicated regression test pins it.

## Summary by CodeRabbit

* **New Features**
  * Added compound `reactive_show` visibility rules (`all`/`any`) spanning multiple fields.
  * Added numeric threshold predicates (`gte`, `gt`, `lte`, `lt`) for value-based showing.
  * Expanded conditional fieldset examples with “International address” and “Bulk surcharge” panels.
* **Bug Fixes**
  * Improved fail-closed behavior for malformed predicates and non-numeric comparisons (safe hidden defaults).
  * Preserved backward compatibility for existing single-field `reactive_show` usage.
* **Documentation**
  * Updated `reactive_show` docs to describe compound and numeric predicate behavior.
* **Tests**
  * Added coverage for numeric thresholds, compound logic, and edge cases.

### CODE REVIEW

File: lib/phlex/reactive/component/helpers.rb [hunk 1]
```diff
@@ -734,6 +775,96 @@ def normalize_show_predicate(predicates, context)
...
+              "got #{connectives.keys.inspect}"
+          end
+
+          connective, terms = connectives.first
+          attrs = options.except(*SHOW_CONNECTIVE_KEYS)
+          unless terms.is_a?(Array) && terms.any?
+            raise ArgumentError,
+              "reactive_show #{connective}: needs at least one term " \
+              "({ field: …, equals:/not:/in:/gte:/… }), got #{terms.inspect}"
+          end
```
[the remark this answers is not in the document]
Comment (mhenrixon) [reply lib/phlex/reactive/component/helpers.rb:857]: Fixed in 8674b3e. Confirmed the leak — `reactive_show(all: [...], equals: "x")` was rendering `equals="x"` as a literal HTML attribute. Rather than silently stripping `SHOW_PREDICATE_KEYS` (which hides the caller mistake), the compound path now raises: a top-level predicate beside a connective is always a misuse — predicates belong inside each term. This matches the single-field path's default-deny posture (a dead/malformed binding fails at render, not silently in the browser). Added a render-level regression asserting no stray predicate attrs plus the raise.
